packages/flutter/lib/src/material/typography.dart

```diff
@@ -269,11 +296,27 @@ class TextTheme {
 ///  * [ThemeData]
 ///  * <http://material.google.com/style/typography.html>
 class Typography {
-  Typography._();
-
   /// A material design text theme with dark glyphs.
-  static const TextTheme black = const TextTheme._black();
+  final TextTheme black;
 
   /// A material design text theme with light glyphs.
-  static const TextTheme white = const TextTheme._white();
+  final TextTheme white;
+
+  const Typography._(this.black, this.white);
+
+  /// Returns the default typography for the specified platform.
+  static Typography forPlatform(TargetPlatform platform) {
```

Comment:
(the rule is "code should be in the order it's run", to make it easier to read. The constructors are run before you do anything with fields, hence they go first.)